### PR TITLE
Add Span Probe instrumentation for method probe

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Add object model into bootstrap for abstracting span API manipulation
Add Span Probe instrumentation for method probe

# Motivation
generate spans for span probes

# Additional Notes

### CODE REVIEW

**Review by ojung — APPROVED** [review]
Looking great, thanks! 👍

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTracer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    AgentTracer.TracerAPI tracerAPI = AgentTracer.get();
+    if (tracerAPI == null) {
+      return DebuggerSpan.NOOP_SPAN;
+    }
+    AgentSpan dynamicSpan = tracerAPI.startSpan(operationName);
+    if (tags != null) {
+      for (String tag : tags) {
+        int idx = tag.indexOf(':');
+        if (idx == -1) {
+          continue;
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTracer.java:22]: ah yes good catch, fixing it

[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTracer.java:22]: looking closely:
Span API use key and value separated, and a null value seems to remove the tag.
also from the doc, span-tags are described by `KEY:VALUE` pair https://docs.datadoghq.com/tracing/guide/add_span_md_and_graph_it/

[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerTracer.java:22]: I will merge it like this. will fix it later if we are stumble upon a case that requires to support value-only

**Pull request merged**